akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -23,19 +23,20 @@ import akka.util.ccompat._
  */
 @InternalApi
 @ccompatUsedUntil213
-private[cluster] final class ClusterHeartbeatReceiver extends Actor with ActorLogging {
+private[cluster] final class ClusterHeartbeatReceiver(getCluster: () => Cluster) extends Actor with ActorLogging {
   import ClusterHeartbeatSender._
 
   // Important - don't use Cluster(context.system) in constructor because that would
   // cause deadlock. See startup sequence in ClusterDaemon.
-  lazy val cluster = Cluster(context.system)
-  lazy val selfHeartbeatRsp = HeartbeatRsp(cluster.selfUniqueAddress)
+  lazy val cluster = getCluster()
+
   lazy val verboseHeartbeat = cluster.settings.Debug.VerboseHeartbeatLogging
 
-  def receive = {
-    case Heartbeat(from) =>
-      if (verboseHeartbeat) cluster.ClusterLogger.logDebug("Heartbeat from [{}]", from)
-      sender() ! selfHeartbeatRsp
+  def receive: Receive = {
+    case hb: Heartbeat =>
+      // TODO log the sequence nr once serialiser is enabled
+      if (verboseHeartbeat) cluster.ClusterLogger.logDebug("Heartbeat from [{}]", hb.from)
+      sender() ! HeartbeatRsp(cluster.selfUniqueAddress, hb.sequenceNr, hb.sendTimeNanos)
```

Comment:
rename to `sendNanoTime` for symmetry with `System.nanoTime`

Reply:
The notion of send time over event time (event creation time) seems odd. Could we consider retaining `NanoTime` and using `eventNanoTime`, and the receiver notes the `processingNanoTime`, it is a more standard notion IMO.